pkg/detectors/falsepositives.go

```diff
@@ -64,16 +81,19 @@ func GetFalsePositiveCheck(detector Detector) func(Result) (bool, string) {
 //
 // Currently, this includes: english word in key or matches common example patterns.
 // Only the secret key material should be passed into this function
-func IsKnownFalsePositive(match string, falsePositives []FalsePositive, wordCheck bool) (bool, string) {
+func IsKnownFalsePositive(match string, falsePositives map[FalsePositive]struct{}, wordCheck bool) (bool, string) {
 	if !utf8.ValidString(match) {
 		return true, "invalid utf8"
 	}
 	lower := strings.ToLower(match)
-	for _, fp := range falsePositives {
+	for fp := range falsePositives {
 		fps := string(fp)
-		if strings.Contains(lower, fps) {
+		if lower == fps {
```

Comment:
You still do exact matching by iterating over `falsePositives` - do you actually use your new map as a map anywhere?

Reply:
That makes sense to me.